### PR TITLE
Change native funcol inductor tests to use fake pg

### PR DESCRIPTION
**Pull request by yifuwang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119951
* #120043
* #120042
* #120025
* #119982
* #119950
* #119881
* #119229
* __->__ #119104
* #119103

Summary:
Previously these tests require more than 2 GPUs to run. Changing them to use fake pg so they can run more often.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

File: test/distributed/test_c10d_functional_native.py [hunk 1]
```diff
@@ -326,23 +330,44 @@ def test_all_to_all_single(self) -> None:
...
-    @torch._inductor.config.patch(debug=True)
     @fresh_inductor_cache()
     def test_inductor_all_reduce_single(self):
-        self._init_process_group()
-
         def func(arg: torch.Tensor) -> torch.Tensor:
             buf0 = arg + 42
             # Expect in-place with inductor allocated buf
-            ar0 = funcol.all_reduce(buf0, "avg", "default")
+            ar0 = funcol.all_reduce(buf0, "avg", "0")
```
Comment (wconstab) [inline test/distributed/test_c10d_functional_native.py:363]: if you use a fakePG as your only pg does "default" no longer map to it?

File: test/distributed/test_c10d_functional_native.py [hunk 2]
```diff
@@ -84,7 +88,7 @@ def _init_process_group(self) -> None:
         torch._C._distributed_c10d._register_process_group("default", dist.group.WORLD)
 
     @skip_if_lt_x_gpu(2)
```
Comment (wconstab) [inline test/distributed/test_c10d_functional_native.py:90]: dont you need to kill the skip_if_lt_x_gpu(2) as well to get what you want about not requiring 2GPUs in ci?

File: test/distributed/test_c10d_functional_native.py [hunk 3]
```diff
@@ -84,7 +88,7 @@ def _init_process_group(self) -> None:
...
-    def test_all_reduce(self) -> None:
+    def test_all_reduce_single(self) -> None:
```
Comment (wanchaol) [inline test/distributed/test_c10d_functional_native.py:91]: Curious what is `all_reduce_single` mean?

File: test/distributed/test_c10d_functional_native.py (hunk 2 above)
Comment (yifuwang) [reply test/distributed/test_c10d_functional_native.py:90]: Oh I'm only changing the Inductor tests to use fake pg. These tests here use real nccl pg.

File: test/distributed/test_c10d_functional_native.py (hunk 3 above)
Comment (yifuwang) [reply test/distributed/test_c10d_functional_native.py:91]: It just means that it's not coalesced. The purpose is to make pattern matching the test easier with `-k`.

File: test/distributed/test_c10d_functional_native.py (hunk 1 above)
Comment (yifuwang) [reply test/distributed/test_c10d_functional_native.py:363]: Preivously, this test explicitly registered the global pg under the name "default" (this was before auto name registration was implemented so I had to bootstrap). By default, the global pg gets the name "0" (this something we could discuss and change).

**Review by wconstab — APPROVED** [review]

**Pull request closed without merging**